snuba/datasets/table_storage.py

```diff
@@ -124,25 +131,25 @@ def get_writer(
         self, options=None, table_name=None, rapidjson_serialize=False
     ) -> BatchWriter:
         from snuba import settings
-        from snuba.clickhouse.http import HTTPBatchWriter
 
         def default(value):
             if isinstance(value, datetime):
                 return value.strftime(DATETIME_FORMAT)
             else:
                 raise TypeError
 
-        return HTTPBatchWriter(
-            self.__table_schema,
-            settings.CLICKHOUSE_HOST,
-            settings.CLICKHOUSE_HTTP_PORT,
+        table_name = table_name or self.__table_schema.get_table_name()
+
+        options = self.__update_writer_options(options)
```

Comment:
Doesn't mypy complain about this ? we are changing the type of options and we are passing something to update_writer_options that does not have the right type.
Could you please apply the type hint to options in the signature of this method ?

Reply:
It doesn't complain, isn't the type inferred from the return value of `self.__update_writer_options`